### PR TITLE
For #58: created File.Asto implementation and tests

### PR DESCRIPTION
**Pull request by paulodamaso**
For #58:
- Created `File.Asto` skeleton implementation
- Created tests for `File.Asto`
- Left puzzle for implementing the actual class

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#61](https://github.com/artipie/maven-adapter/issues/61) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/40d3901a-7733-4c07-a06c-44d761db3c6e); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by Vatavuk** [comment]
@g4s8 This PR contains an unimplemented class and two disabled test. Are you ok with this "TDD" approach?  To me it seems that it adds no significant value and I would prefer one method/test over "ignored" tests only approach. This is the second PR done in this manner so I'm asking for future reference. I'm ok with whatever you decide.

File: src/main/java/com/artipie/maven/file/File.java [hunk 1]
```diff
@@ -50,4 +52,43 @@
...
+         * @param key Asto key.
+         * @param storage Asto storage.
+         */
+        Asto(final Key key, final Storage storage) {
+            this.key = key;
+            this.storage = storage;
+        }
+
+        @Override
+        public byte[] content() {
```
Comment (g4s8) [inline src/main/java/com/artipie/maven/file/File.java]: @paulodamaso I don't think the method signature is correct. We're using `Publisher<ByteBuffer>` for reading and writing data to and from `Storage` interface and to accept and respond for HTTP requests. Declaring the `byte[]` signature means that we should block the thread until read all the data from storage resource, it's not acceptable in paradigm of reactive approach.

**Comment by g4s8** [comment]
> @g4s8 This PR contains an unimplemented class and two disabled tests. Are you ok with this "TDD" approach? To me it seems that it adds no significant value and I would prefer one method/test over "ignored" tests only approach. This is the second PR done in this manner so I'm asking for future reference.

@Vatavuk @paulodamaso I think @Vatavuk is right here: at the beginning the new features are more important for us than unit tests, so first we need to define some core classes and add tests later. I'd suggest to use integration tests for early stages, just to be sure that whole module works, but avoid unit tests, since the code base is volatile and many classes may change at any moment.

**Comment by paulodamaso** [comment]
@g4s8 Changes made, please take a look. @Vatavuk I'll use your suggestion on the next issues, thanks

**Comment by Vatavuk** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/maven-adapter/issues/61#issuecomment-605823039))

@Vatavuk This is what I know about this job in CT2E6TK9B, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#61](https://github.com/artipie/maven-adapter/issues/61) is [in scope](http://www.zerocracy.com/policy.html#14) for 4days
  * The role is `REV`
  * The job is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk) for [4days](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @vatavuk[/z](https://www.0crat.com/u/vatavuk) will get 25% of their hourly rate for this job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @paulodamaso[/z](https://www.0crat.com/u/paulodamaso): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Fmaven-adapter%2361%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by g4s8** [comment]
@0crat out

**Comment by 0crat** [comment]
Job audit: performer @vatavuk[/z](https://www.0crat.com/u/vatavuk)[/z](https://www.0crat.com/u/vatavuk) didn't make CR comments

**Pull request merged**